cross_compile/sysroot_compiler.py

```diff
@@ -219,34 +222,50 @@ def _setup_sysroot_dir(self, custom_script: Optional[str]) -> None:
         'sysroot' directory in order to copy the assets to it.
         See https://docs.docker.com/engine/reference/builder/#copy
         """
-        logger.info('Checking sysroot directory...')
         if not self._target_sysroot.exists():
             raise FileNotFoundError(SYSROOT_NOT_FOUND_ERROR_STRING.substitute(
                 sysroot_dir=self._target_sysroot))
-
         logger.debug('Sysroot directory exists.')
+
         if not self._ros_workspace_dir.exists():
             raise FileNotFoundError(ROS_WS_DIR_ERROR_STRING.substitute(
                 ros_ws=self._ros_workspace_dir))
         logger.debug('ROS workspace exists.')
+
         if not self._qemu_directory.exists():
             raise FileNotFoundError(
                 QEMU_DIR_ERROR_STRING.substitute(qemu_dir=QEMU_DIR_NAME))
         if not os.listdir(str(self._qemu_directory.absolute())):
             raise FileNotFoundError(
                 QEMU_EMPTY_ERROR_STRING.substitute(qemu_dir=QEMU_DIR_NAME))
         logger.debug('QEMU binaries exist')
+
         shutil.copy(
             str(self._dockerfile_directory), str(self._target_sysroot))
         if not self._expected_dockerfile_directory.exists():
             raise FileNotFoundError(COPY_DOCKER_WS_ERROR_STRING.substitute(
                 dockerfile=ROS_DOCKERFILE_NAME))
+        logger.debug('Copied Dockerfile')
+
+        custom_data_dest = str(self._target_sysroot / 'user-custom-data')
+        shutil.rmtree(
+            custom_data_dest, ignore_errors=True
```

Comment:
Instead of converting to string, would [`as_posix()`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_posix) or `as_uri()` make more sense?

Reply:
There's a preexisting pattern of conversion to string in the codebase so I'm hesitant to do something different here. We may not be running on a posix platform so i'm not sure that particular way would be the most correct.